src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -127,16 +134,34 @@ public void deleteAssignment(@PathVariable("assignmentId") int assignmentId) {
     @GetMapping("/assignments/{assignmentId}/grades")
     public List<GradeDTO> getAssignmentGrades(@PathVariable("assignmentId") int assignmentId) {
 
-        // TODO remove the following line when done
 
         // get the list of enrollments for the section related to this assignment.
-		// hint: use te enrollment repository method findEnrollmentsBySectionOrderByStudentName.
+		// hint: use te enrollment repository method findEnrollmentsBySectionOrderByStudentName
+        Assignment a = assignmentRepository.findById(assignmentId).orElse(null);
+        if(a == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "assignment not found" + assignmentId);
+        }
+        int secNo = a.getSection().getSectionNo();
+        List<Enrollment> enrollments = enrollmentRepository.findEnrollmentsBySectionNoOrderByStudentName(secNo);
         // for each enrollment, get the grade related to the assignment and enrollment
 		//   hint: use the gradeRepository findByEnrollmentIdAndAssignmentId method.
         //   if the grade does not exist, create a grade entity and set the score to NULL
         //   and then save the new entity
+        List<GradeDTO> gradeDTOs = new ArrayList<>();
+        for (Enrollment e : enrollments) {
+            Grade g = gradeRepository.findByEnrollmentIdAndAssignmentId(e.getEnrollmentId(), assignmentId);
+            gradeDTOs.add(new GradeDTO(
```

Comment:
Need to implement the getters and setters in the Grade.java class; they currently don't exist hence this code will not work.